examples/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Viam Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+add_subdirectory(dial)
```

Comment:
The `examples` should either be pushed under `src`, or have their own CMake infra that builds against the installed SDK. That's useful because it provides both an example of how to build against the SDK, and can be used as part of the CI pipeline to validate that the produced SDK can be built against.

Reply:
Since we are in agreement on the latter approach, I've added top-level TODO to that effect.

Reply:
I actually did this slightly differently. The examples are still part of the main SDK build, but they are *also* referenced by standalone CMakeList.txt and Makefile examples under the new `examples/project` tree. That ensures that the examples don't break when working in the SDK, but also re-use them as part of the project examples, which we can also build in CI.